src/main/java/org/fogbowcloud/sebal/parsers/WeatherStation.java

```diff
@@ -140,136 +83,119 @@ protected String readFullRecord(Date date, List<JSONObject> stations, int number
-						.optString(SEBALAppConstants.JSON_MAX_TEMPERATURE) == null) {
-			maxTemp = "NA";
-		}
-		if (closestRecord.optString(SEBALAppConstants.JSON_SOLAR_RADIATION)
-				.isEmpty()
-				|| closestRecord
-						.optString(SEBALAppConstants.JSON_SOLAR_RADIATION) == null) {
-			solarRad = "NA";
+	protected boolean hasRecord(JSONArray stationData, String key, String value) {
+		boolean result = false;
+		for (int i = 0; i < stationData.length() && !result; i++) {
+			JSONObject stationDataRecord = stationData.optJSONObject(i);
+			
+			if (stationDataRecord.optString(key).equals(value)) {
+				result = true;
+			}
 		}
-
-		toReturn.append(stationId + ";" + dateValue + ";" + timeValue + ";"
-				+ latitude + ";" + longitude + ";" + windSpeed + ";" + airTemp
-				+ ";" + dewTemp + ";" + avgAirTemp + ";" + relativeHumidity
-				+ ";" + minTemp + ";" + maxTemp + ";" + solarRad + ";\n");
+		return result;
 	}
 
-	public double zx(double lat, double lon) {
-		if (properties.getProperty("altitude_sensor_velocidade") != null) {
-			return Double.parseDouble(properties
-					.getProperty("altitude_sensor_velocidade"));
+	protected void windSpeedCorrection(JSONArray stationData) {
+		if (stationData != null) {
+			for (int i = 0; i < stationData.length(); i++) {
+				JSONObject stationDataRecord = stationData.optJSONObject(i);
+
+				if (Double.parseDouble(stationDataRecord
+						.optString(SEBALAppConstants.JSON_STATION_WIND_SPEED)) < Double
+								.parseDouble(WeatherStation.MINIMUM_WIND_SPEED_VALUE)) {
+
+					stationDataRecord.remove(SEBALAppConstants.JSON_STATION_WIND_SPEED);
+					stationDataRecord.put(SEBALAppConstants.JSON_STATION_WIND_SPEED,
+							WeatherStation.MINIMUM_WIND_SPEED_VALUE);
+				}
+			}
 		}
-		return 6.;
 	}
 
-	public double d(double lat, double lon) {
-		return 4. * 2 / 3;
+	protected boolean stationContainsAll(JSONObject station) {
+		return !station.optString(SEBALAppConstants.JSON_STATION_DATE).isEmpty()
+				&& !station.optString(SEBALAppConstants.JSON_STATION_TIME).isEmpty()
+				&& !station.optString(SEBALAppConstants.JSON_STATION_LATITUDE).isEmpty()
+				&& !station.optString(SEBALAppConstants.JSON_STATION_LONGITUDE).isEmpty()
+				&& !station.optString(SEBALAppConstants.JSON_AIR_TEMPERATURE).isEmpty()
+				&& !station.optString(SEBALAppConstants.JSON_DEWPOINT_TEMPERATURE).isEmpty()
+				&& !station.optString(SEBALAppConstants.JSON_STATION_WIND_SPEED).isEmpty()
+				&& Double.parseDouble(
+						station.optString(SEBALAppConstants.JSON_STATION_WIND_SPEED)) >= 0.3;
 	}
 
-	public double hc(double lat, double lon) {
-		if (properties.getProperty("hc") != null) {
-			return Double.parseDouble(properties.getProperty("hc"));
+	private String generateStationData(JSONArray stationData) {
+		StringBuilder result = new StringBuilder();
+		for (int i = 0; i < stationData.length(); i++) {
+			result.append(checkVariablesAndBuildString(stationData.optJSONObject(i)));
 		}
-		return 4.0;
+		return result.toString().trim();
 	}
 
```

Comment:
usually, methods that have a And are odd. the idea it to KISS.

Reply:
Done, review!